src/librustc_errors/diagnostic.rs

```diff
@@ -191,11 +197,23 @@ impl Diagnostic {
            message: &str,
            span: MultiSpan,
            render_span: Option<RenderSpan>) {
+        self.sub_with_list(level, message, span, render_span, vec![]);
```

Comment:
@nikomatsakis,
Is `vec![]` idiomatic way to create an empty vector?
I mean that `vec!` is a macro.
It requires expansion.
I think that `Vec::new` is more appropriate.